backend/src/graphql/resolver/util/creations.test.ts

```diff
@@ -0,0 +1,268 @@
+          memo: 'Contribution for the last month',
+          creationDate: contributionDateFormatter(
+            new Date(now.getFullYear(), now.getMonth() - 1, now.getDate()),
+          ),
+        },
+      })
+    })
+
+    it('has the correct data setup', async () => {
+      await expect(Contribution.find()).resolves.toEqual([
+        expect.objectContaining({
+          userId: user.id,
+          contributionDate: setZeroHours(now),
+          amount: expect.decimalEqual(250),
+          memo: 'Admin contribution for this month',
+          moderatorId: admin.id,
+          contributionType: 'ADMIN',
+          contributionStatus: 'PENDING',
+        }),
+        expect.objectContaining({
+          userId: user.id,
+          contributionDate: setZeroHours(
+            new Date(now.getFullYear(), now.getMonth() - 1, now.getDate()),
+          ),
+          amount: expect.decimalEqual(160),
+          memo: 'Admin contribution for the last month',
+          moderatorId: admin.id,
+          contributionType: 'ADMIN',
+          contributionStatus: 'PENDING',
+        }),
+        expect.objectContaining({
+          userId: user.id,
+          contributionDate: setZeroHours(
+            new Date(now.getFullYear(), now.getMonth() - 2, now.getDate()),
+          ),
+          amount: expect.decimalEqual(450),
+          memo: 'Admin contribution for two months ago',
+          moderatorId: admin.id,
+          contributionType: 'ADMIN',
+          contributionStatus: 'PENDING',
+        }),
+        expect.objectContaining({
+          userId: user.id,
+          contributionDate: setZeroHours(now),
+          amount: expect.decimalEqual(400),
+          memo: 'Contribution for this month',
+          moderatorId: null,
+          contributionType: 'USER',
+          contributionStatus: 'PENDING',
+        }),
+        expect.objectContaining({
+          userId: user.id,
+          contributionDate: setZeroHours(
+            new Date(now.getFullYear(), now.getMonth() - 1, now.getDate()),
+          ),
+          amount: expect.decimalEqual(500),
+          memo: 'Contribution for the last month',
+          moderatorId: null,
+          contributionType: 'USER',
+          contributionStatus: 'PENDING',
+        }),
+      ])
+    })
+
+    describe('call getUserCreation now', () => {
+      it('returns the expected open contributions', async () => {
+        await expect(getUserCreation(user.id, 0)).resolves.toEqual([
+          expect.decimalEqual(550),
+          expect.decimalEqual(340),
+          expect.decimalEqual(350),
+        ])
+      })
+
+      describe('run forward in time one hour before next month', () => {
+        const targetDate = new Date(now.getFullYear(), now.getMonth() + 1, 0, 23, 0, 0)
+
+        beforeAll(() => {
+          jest.useFakeTimers()
+          /* eslint-disable-next-line @typescript-eslint/no-empty-function */
+          setTimeout(() => {}, targetDate.getTime() - now.getTime())
```

Comment:
What happen when targertDate.Time is lower than now.Time?

Reply:
Godd point. This unit test is flaky every last hour of the month. I accept this.